src/Core/MySQL/PacketsProtocolText.cpp

```diff
@@ -173,30 +185,52 @@ ColumnDefinition getColumnDefinition(const String & column_name, const TypeIndex
         case TypeIndex::Float32:
             column_type = ColumnType::MYSQL_TYPE_FLOAT;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
+            decimals = 31;
             break;
         case TypeIndex::Float64:
             column_type = ColumnType::MYSQL_TYPE_DOUBLE;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
+            decimals = 31;
             break;
         case TypeIndex::Date:
+        case TypeIndex::Date32:
             column_type = ColumnType::MYSQL_TYPE_DATE;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
             break;
         case TypeIndex::DateTime:
             column_type = ColumnType::MYSQL_TYPE_DATETIME;
             flags = ColumnDefinitionFlags::BINARY_FLAG;
             break;
-        case TypeIndex::String:
-        case TypeIndex::FixedString:
-            column_type = ColumnType::MYSQL_TYPE_STRING;
-            charset = CharacterSet::utf8_general_ci;
+        case TypeIndex::DateTime64:
+            column_type = ColumnType::MYSQL_TYPE_DATETIME;
+            flags = ColumnDefinitionFlags::BINARY_FLAG;
+            break;
+        case TypeIndex::Decimal32:
+        case TypeIndex::Decimal64:
+            column_type = ColumnType::MYSQL_TYPE_DECIMAL;
+            flags = ColumnDefinitionFlags::BINARY_FLAG;
+            break;
+        case TypeIndex::Decimal128: {
+            // MySQL Decimal has max 65 precision and 30 scale
```

Comment:
I find proof of "max 65 precision" [here](https://dev.mysql.com/doc/refman/8.0/en/fixed-point-types.html) but the same page doesn't say the max scale is 30. Is there documented evidence of the latter?